tests/ui/associated-type-bounds/issue-102335-ty.stderr

```diff
@@ -3,6 +3,11 @@ error[E0229]: associated type bindings are not allowed here
    |
 LL |     type A: S<C<i32 = u32> = ()>;
    |                 ^^^^^^^^^ associated type not allowed here
+   |
+help: to use `u32` as a generic argument specify it directly
```

Comment:
I think we should use the wording “you might've meant to write a type parameter here” since we don't actually know if the container type has any args. Well, you could look at the generics of it with `generics_of`.

Reply:
Now the code checks if the container type has generic params using `generics_of` as you suggested. If it doesn't have any params, then we show a different message and suggestion. In the other case I have retained the original message `to use ... as a generic argument specify it directly` because now we can suggest that with confidence. However, I'm open to changing this text if it's still not appropriate. Let me know if you want that.